slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -143,9 +144,10 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
     // Connection settings
     hconf.setDataSourceClassName(c.getStringOr("dataSourceClass", null))
     Option(c.getStringOr("driverClassName", c.getStringOr("driver"))).map(hconf.setDriverClassName _)
-    hconf.setJdbcUrl(c.getStringOr("url", null))
-    c.getStringOpt("user").foreach(hconf.setUsername)
-    c.getStringOpt("password").foreach(hconf.setPassword)
+    hconf.setJdbcUrl(parseDatabaseUrl(c, (user, password) => {
+      user.foreach(hconf.setUsername)
+      password.foreach(hconf.setPassword)
+    }))
```

Comment:
If I am not mistaken this breaks existing projects in that it doesn't user the username config field anymore. If this is accepted it should be allowing the old style but deprecating it.